src/app/fdctl/run/tiles/fd_cswtch.c

```diff
@@ -0,0 +1,197 @@
+#include "../../../../disco/tiles.h"
+
+#include "generated/cswtch_seccomp.h"
+
+#include <unistd.h>
+#include <fcntl.h>
+#include <errno.h>
+
+#define REPORT_INTERVAL_MILLIS (100L)
+
+typedef struct {
+  long next_report_nanos;
+
+  ulong   tile_cnt;
+  int     status_fds[ FD_TILE_MAX ];
+  ulong * metrics[ FD_TILE_MAX ];
+} fd_cswtch_ctx_t;
+
+FD_FN_CONST static inline ulong
+scratch_align( void ) {
+  return 128UL;
+}
+
+FD_FN_PURE static inline ulong
+scratch_footprint( fd_topo_tile_t const * tile ) {
+  (void)tile;
+  ulong l = FD_LAYOUT_INIT;
+  l = FD_LAYOUT_APPEND( l, alignof( fd_cswtch_ctx_t ), sizeof( fd_cswtch_ctx_t ) );
+  return FD_LAYOUT_FINI( l, scratch_align() );
+}
+
+FD_FN_CONST static inline void *
+mux_ctx( void * scratch ) {
+  return (void*)fd_ulong_align_up( (ulong)scratch, alignof( fd_cswtch_ctx_t ) );
+}
+
+static void
+privileged_init( fd_topo_t *      topo,
+                 fd_topo_tile_t * tile,
+                 void *           scratch ) {
+  (void)tile;
+
+  FD_SCRATCH_ALLOC_INIT( l, scratch );
+  fd_cswtch_ctx_t * ctx = FD_SCRATCH_ALLOC_APPEND( l, alignof( fd_cswtch_ctx_t ), sizeof( fd_cswtch_ctx_t ) );
+
+  FD_TEST( topo->tile_cnt<FD_TOPO_MAX_TILES );
+
+  ctx->tile_cnt = topo->tile_cnt;
+  for( ulong i=0UL; i<topo->tile_cnt; i++ ) {
+    ulong * metrics = fd_metrics_join( fd_topo_obj_laddr( topo, topo->tiles[ i ].metrics_obj_id ) );
+
+    for(;;) {
+      ulong pid, tid;
+      if( FD_UNLIKELY( tile->id==i ) ) {
+        pid = fd_sandbox_getpid();
+        tid = fd_sandbox_gettid();
+      } else {
+        pid = fd_metrics_tile( metrics )[ FD_METRICS_GAUGE_TILE_PID_OFF ];
+        tid = fd_metrics_tile( metrics )[ FD_METRICS_GAUGE_TILE_TID_OFF ];
+        if( FD_UNLIKELY( !pid || !tid ) ) {
+          FD_SPIN_PAUSE();
+          continue;
+        }
+      }
+
+      char path[ 64 ];
+      FD_TEST( fd_cstr_printf_check( path, sizeof( path ), NULL, "/proc/%lu/task/%lu/status", pid, tid ) );
+      ctx->status_fds[ i ] = open( path, O_RDONLY );
+      ctx->metrics[ i ] = fd_metrics_tile( metrics );
+      if( FD_UNLIKELY( -1==ctx->status_fds[ i ] ) ) FD_LOG_ERR(( "open failed (%i-%s)", errno, strerror( errno ) ));
```

Comment:
It seems like `/proc` [can be mounted with various options](https://docs.kernel.org/filesystems/proc.html#chapter-4-configuring-procfs) that may make this unreadable. We might want to have an option to disable it or to fail more gracefully.

Reply:
We can add that if it is a problem in practice, I don't think it's very likely anyone remounts /proc with weird options.